cpp/log/cert_checker.cc

```diff
@@ -178,12 +179,15 @@ Status CertChecker::CheckIssuerChain(CertChain* chain) const {
 Status CertChecker::CheckPreCertChain(PreCertChain* chain,
                                       string* issuer_key_hash,
                                       string* tbs_certificate) const {
-  if (!chain || !chain->IsLoaded())
+  if (!chain || !chain->IsLoaded()) {
     return Status(util::error::INVALID_ARGUMENT, "invalid certificate chain");
-  Cert::Status status = chain->IsWellFormed();
-  if (status == Cert::FALSE)
+  }
+
+  StatusOr<bool> status = chain->IsWellFormed();
+  if (status.ok() && !status.ValueOrDie()) {
     return Status(util::error::INVALID_ARGUMENT, "prechain not well formed");
-  if (status != Cert::TRUE) {
+  }
+  if (!status.ok() || !status.ValueOrDie()) {
```

Comment:
Not sure about that one. ValueOrDie() only dies if status is not OK. We want to return an error status if the chain isn't well formed.